src/foraging_gui/Foraging.py

```diff
@@ -647,13 +647,8 @@ def _OpenBonsaiWorkflow(self,runworkflow=1):
 
         SettingsBox = 'Settings_box{}.csv'.format(self.tower_number)
         CWD=os.path.join(os.path.dirname(os.getcwd()),'workflows')
-        #if len(sys.argv) == 1:
-        #    subprocess.Popen(self.bonsai_path+' '+self.bonsaiworkflow_path+' -p '+'SettingsPath='+self.SettingFolder+'\\'+SettingsBox+ ' --start',cwd=CWD)
-        #else:
-        #    if self.tower_number==1:
-        #        subprocess.Popen(self.bonsai_path+' '+self.bonsaiworkflow_path+' -p '+'SettingsPath='+self.SettingFolder+'\\'+SettingsBox,cwd=CWD)
-        #    else:
-        subprocess.Popen(self.bonsai_path+' '+self.bonsaiworkflow_path+' -p '+'SettingsPath='+self.SettingFolder+'\\'+SettingsBox+ ' --start',cwd=CWD)
+        subprocess.Popen(self.bonsai_path+' '+self.bonsaiworkflow_path+' -p '+'SettingsPath='+self.SettingFolder+'\\'+SettingsBox+ ' --start',cwd=CWD,shell=True)
```

Comment:
Can you set the shell==False for Tower_EphysRig1, Tower_EphysRig2 and Tower_EphysRig3? I need to open the bonsai console in the ephys rig.

Reply:
How do you start the GUI on those rigs? Do you use a `.bat` file? Or do you start it in a console window?

I think that if you start the GUI in a console window or a `.bat` file with `python Foraging.py` then the console will open regardless of whether `shell=True` here. I think `shell=False` only opens a console if one is not already open. I could be wrong though. Can you test it on one of those ephys rigs? I dont know what to look for in the console as Bonsai output

Reply:
@XX-Yin I just tested this on my laptop by trying to start bonsai with a box number that doesn't exist on my computer, and with `python Foraging.py` the bonsai errors are displayed in the console, like they were previously. I believe this will solve your use case, but I'd like to hear from you about how you start the GUI, and whether you can still see the video output

Reply:
@alexpiet In the ephys rig, I also started GUI through a `.bat` file (Previously, I used `start python Foraging.py` instead of `start python Foraging.py 1`).  After your modification, I believe I should change to use `start python Foraging.py 1`

Reply:
Either `start python Foraging.py` or `start python Foraging.py 1` should work. The box number defaults to 1 if it is not passed in. Did you get a chance to try this out on your ephys rig?

Reply:
I tested on the ephys rig. It works.